packages/diagrams/lib/src/text.dart

```diff
@@ -0,0 +1,204 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+import 'package:diagram_capture/diagram_capture.dart';
+import 'package:flutter/material.dart';
+import 'diagram_step.dart';
+
+class TextDiagram extends StatelessWidget implements DiagramMetadata {
+  const TextDiagram(this.name);
+
+  @override
+  final String name;
+
+  @override
+  Widget build(BuildContext context) {
+    final TextPainter textPainter = TextPainter(
+      text: const TextSpan(
+        text:' Aalg我हिन्दी ',
+        style: TextStyle(
+          fontSize: 100,
+          color: Colors.black,
+        ),
+      ),
+      textAlign: TextAlign.left,
+      textDirection: TextDirection.ltr,
+    );
+    switch (name) {
+      case 'text_height_diagram':
+        return Container(
+          width: 800,
+          height: 200,
+          color: Colors.white,
+          child: CustomPaint(
+            size: Size(1000, 300),
+            painter: CurvePainter(textPainter),
+          ),
+        );
+        break;
+      default:
+        return const Text('Error');
+        break;
+    }
+  }
+}
+
+class CurvePainter extends CustomPainter {
+
+  const CurvePainter(this.textPainter);
+
+  final TextPainter textPainter;
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    textPainter.layout();
+
+    final List<TextBox> boxes = textPainter.getBoxesForSelection(
+      const TextSelection(baseOffset: 0, extentOffset: 99)
```

Comment:
What's the 99 magic number?

Reply:
It just needs to be a large number to ensure it extends out beyond the end of short string.